app/lib/features/tasks/sheets/create_update_task_item.dart

```diff
@@ -0,0 +1,228 @@
+              Text(
+                L10n.of(context).addTask,
+                textAlign: TextAlign.center,
+                style: Theme.of(context).textTheme.titleMedium,
+              ),
+              const SizedBox(height: 20),
+              _widgetTaskName(),
+              const SizedBox(height: 20),
+              _widgetDescriptionName(),
+              const SizedBox(height: 20),
+              _widgetDueDate(),
+              const SizedBox(height: 20),
+              _widgetAddButton(),
+              const SizedBox(height: 20),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _widgetTaskName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).taskName,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          autofocus: true,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).name,
+          ),
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          controller: _taskNameController,
+          validator: (value) => (value?.isNotEmpty == true)
+              ? null
+              : L10n.of(context).pleaseEnterAName,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDescriptionName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).description,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          decoration: InputDecoration(
+            hintText: L10n.of(context).description,
+          ),
+          minLines: 4,
+          maxLines: 4,
+          controller: _taskDescriptionController,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDueDate() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).dueDate,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          readOnly: true,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).dueDate,
+            suffixIcon: IconButton(
+              onPressed: selectDueDate,
+              icon: const Icon(Icons.calendar_month),
+            ),
+          ),
+          onTap: selectDueDate,
+          controller: _taskDueDateController,
+        ),
+      ],
+    );
+  }
+
+  Future<void> selectDueDate() async {
+    final date = await showDatePicker(
+      context: context,
+      initialDate: selectedDate,
+      firstDate: DateTime.now(),
+      lastDate: DateTime.now().addYears(1),
+    );
+    if (date == null || !mounted) return;
+    selectedDate = date;
+    _taskDueDateController.text = taskDueDateFormat(date);
+    setState(() {});
```

Comment:
seen this empty set-state pattern a few times, but shouldn't this rather be:
```suggestion
    setState(() {
      selectedDate = date;
      _taskDueDateController.text = taskDueDateFormat(date);
    });
```
?